packages/xrpl/test/integration/transactions/trustSet.test.ts

```diff
@@ -85,4 +85,45 @@ describe('TrustSet', function () {
     },
     TIMEOUT,
   )
+
+  it(
+    'Create a Deep-Frozen trust line',
+    async () => {
+      assert(wallet2 != null)
+      // preemptively deep-freeze a trust line with the specified counter-party/currency-code
+      const tx: TrustSet = {
+        TransactionType: 'TrustSet',
+        Account: testContext.wallet.classicAddress,
+        LimitAmount: {
+          currency: 'USD',
+          issuer: wallet2.classicAddress,
+          value: '10',
+        },
+        Flags: {
+          tfSetFreeze: true,
+          tfSetDeepFreeze: true,
+        },
+      }
+
+      const response = await testTransaction(
+        testContext.client,
+        tx,
+        testContext.wallet,
+      )
+      assert.equal(response.result.engine_result, 'tesSUCCESS')
+
+      // assert that the trust line is deep-frozen
+      const trustLine = await testContext.client.request({
+        command: 'account_lines',
+        account: testContext.wallet.classicAddress,
+      })
+
+      // assert that the TrustLine is deep-frozen
+      assert.equal(trustLine.result.lines[0].freeze, true)
+
+      // Keshava: ensure that account_lines RPC response contains a deep_freeze flag
+      // assert.equal(trustLine.result.lines[0].deep_freeze, true)
```

Comment:
alternatively (or do both), you can check for `account_objects` and check manually that the trustline object has toggled the deep freeze flag

Reply:
fixed in fixed in [c9ecd33](https://github.com/XRPLF/xrpl.js/pull/2873/commits/c9ecd33eb88848d25d1dca2a24c5270ac356cb04)